Documented how to build on windows and linux (#135)
githubdoe/DFTFringe: README.md

#### Get minGw

The easiest way I have found to install it from command line requires to have Python installed. It then uses [aqtinstall](https://aqtinstall.readthedocs.io/en/latest/).

From within `C:\buildingDFTFringe` do the following:  
```
pip install aqtinstall
aqt install-tool windows desktop tools_mingw qt.tools.win64_mingw810
```

Add minGW to your Path: `C:\buildingDFTFringe\Tools\mingw810_64\bin`
Add Qmake to your Path: `C:\buildingDFTFringe\5.15.2\mingw81_64\bin`

#### Get QT

Here again, I have found easier to use Python and [aqtinstall](https://aqtinstall.readthedocs.io/en/latest/) to get QT.

From within `C:\buildingDFTFringe` do the following:
```
aqt install-qt windows desktop 5.15.2 win64_mingw81 -m qtcharts qtdatavis3d
aqt install-qt windows desktop 5.15.2 win64_mingw81 --archives qtbase qtsvg
```

#### Get Qt creator

This is not a necessary step but if you are building on your own computer you probably want to debug and need Qt creator IDE.
```
aqt install-tool windows desktop tools_qtcreator
```

### Option B: using QT IDE

> Download the **open source** version of Qt framework https://www.qt.io/download-open-source

Reviewer: This option B section is excellent. You included mention of "archive" and "filter" button which wasn't in my instructions but which I found on the internet somewhere and Sorin confirmed just today.

I think it would be good to mention that using this method you get the debugger built into the IDE, "you can set breakpoints, step through the code, and examine variables".  Although maybe you can also do that in GDB?  It's probably not as user friendly?

Run the installer and go through it.
When it's time to select which components to install in custom installation you must first tick the `archive` checkbox and click on `filter`. Without this, you won't be able to install the old 
Qt 5.15.2 version.  

**Which components to choose ?**  
You will need `QT 5.15.2`.  
For faster installation, you probably do not need `Qt design studio`.   
In the details of Qt 5.15.2 you only need `MinGW 8.1.0 64-bit`, `QT Charts` and `Qt Data Visualization`.  
You probably want to install `Qt Creator`.  
You may want to install `Cmake`, `Ninja` and `Qt Installer Framework`.

### Option C: using Windows Subsystem for Linux (WSL)

Since Windows 10, Windows provides a native Linux subsystem. You don't need a virtual machine or a dual boot and can run any Linux application directly inside Windows. It works also for graphical applications.  
More information about installation are available here: https://learn.microsoft.com/en-us/windows/wsl/install  

All what you need to do is:
```
wsl --install
```

> To run you already installed WSL:

Reviewer: ```suggestion
To run if you already installed WSL:
```

```
wsl
```

> For the resto of the build process you can now relly on [Linux build process](how-to-build-dftfringe-on-linux).

Reviewer: ```suggestion
For the rest of the build process you can now rely on [Linux build process](how-to-build-dftfringe-on-linux).
```

Then from within `C:\buildingDFTFringe\qwt-6.1.6` do the following:  
```
qmake.exe
mingw32-make -j4
```

#### Build Lapack and Blas

Get [Lapack](https://www.netlib.org/lapack/) source code version 3.11.0 in your prefered way (clone the repo, download the archive, homing pigeon, ...) and have it in folder named `C:\buildingDFTFringe\lapack`. Lapack comes with Blas.

Then from within `C:\buildingDFTFringe` do the following:  
```
cmake -G "MinGW Makefiles" -S lapack -B build_lapack -D BUILD_SHARED_LIBS=ON -D CMAKE_SHARED_LINKER_FLAGS="-Wl,--allow-multiple-definition"
cmake --build ./build_lapack -j4
```

#### Build Armadillo

Get [Armadillo](https://arma.sourceforge.net/) source code version 12.6.7 in your prefered way (clone the repo, download the archive, homing pigeon, ...) and have it in folder named `C:\buildingDFTFringe\armadillo-12.6.7`.

Then from within `C:\buildingDFTFringe` do the following:  
```
cmake -D CMAKE_PREFIX_PATH=C:/buildingDFTFringe/build_lapack -G "MinGW Makefiles" -S armadillo-12.6.7 -B build_armadillo
cmake --build ./build_armadillo -j4
```

> It's important that Armadillo known the path to Lapack to work correctly. Here we use CMAKE_PREFIX_PATH but you can also add it to your Path or use other methods.

Reviewer: ```suggestion
It's important that Armadillo knows the path to Lapack to work correctly. Here we use CMAKE_PREFIX_PATH but you can also add it to your Path or use other methods.
```

### Option B: getting DLLs from official installer

> First you need to get the header files for each dependency. Than follow [this](https://github.com/githubdoe/DFTFringe?tab=readme-ov-file#option-c-copying-from-release) to copy the necessary DLLs from release to your build folder.

Reviewer: ```suggestion
First you need to get the header files for each dependency. Then follow [this](https://github.com/githubdoe/DFTFringe?tab=readme-ov-file#option-c-copying-from-release) to copy the necessary DLLs from release to your build folder.
```
